File: packages/google_mlkit_translation/android/src/main/java/com/google_mlkit_translation/GoogleMlKitTranslationPlugin.java
package com.google_mlkit_translation;

import androidx.annotation.NonNull;

import io.flutter.embedding.engine.plugins.FlutterPlugin;

public class GoogleMlKitTranslationPlugin implements FlutterPlugin {
    private TextTranslator translateApi;

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
        translateApi = new TextTranslator();
        Pigeon.OnDeviceTranslatorApi.setUp(
        flutterPluginBinding.getBinaryMessenger(), translateApi);
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
        Pigeon.OnDeviceTranslatorApi.setUp(binding.getBinaryMessenger(), null);
    }
}

File: packages/google_mlkit_translation/android/src/main/kotlin/com/google_mlkit_translation/GoogleMlKitTranslationPlugin.kt
package com.google_mlkit_translation

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodChannel

class GoogleMlKitTranslationPlugin : FlutterPlugin {
    private lateinit var channel: MethodChannel

    companion object {
        private const val CHANNEL_NAME = "google_mlkit_on_device_translator"
    }

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(binding.binaryMessenger, CHANNEL_NAME)
        channel.setMethodCallHandler(TextTranslator())
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
